demo/Dockerfile

```diff
@@ -46,3 +46,12 @@ COPY run-demo.sh /usr/local/bin/run-demo.sh
 ENV JAVA_OPTS -Djenkins.install.state=INITIAL_SECURITY_SETUP
 
 CMD /usr/local/bin/run-demo.sh
+
+RUN \
+    apt-get update && \
+    apt-get install -y apt-transport-https ca-certificates && \
+    apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D && \
+    echo "deb https://apt.dockerproject.org/repo debian-jessie main" > /etc/apt/sources.list.d/docker.list && \
+    apt-get update && \
+    apt-get install -y docker-engine
```

Comment:
Definitely do not want this. If `-v $(shell which docker):/usr/bin/docker` is a problem when it would be better to download just the Docker CLI ([hint](https://github.com/jenkinsci/docker-commons-plugin/blob/9d298f88b30b1510d169d2407a039923a31a663c/src/main/java/org/jenkinsci/plugins/docker/commons/tools/DockerToolInstaller.java#L71-L119)).